### PR TITLE
[Fix] Fix AttributeError in Qwen2.5 LoRA: 'Qwen2ForCausalLM' object has no attribute 'get_hidden_dim'

### PR DESCRIPTION
**Pull request by mssongit**
## Motivation

This PR fixes an `AttributeError` that occurs when using LoRA with the Qwen2.5 model. The error stems from an attempt to call `get_module_name` and `get_hidden_dim` on the `Qwen2ForCausalLM` model, which does not have this attribute. This breaks the LoRA initialization process, specifically when preallocating memory for the LoRA adapters.

## Modifications

Add fallback function for `get_module_name` and `get_hidden_dim`

## Checklist

- [x] Format the code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by merrymercy** [comment]
@mssongit Could you fix the lint error?

**Comment by mssongit** [comment]
> @mssongit Could you fix the lint error?

Sure! I’ve already fixed the lint error by ensuring the code properly checks for the existence of the get_hidden_dim method and falls back to config.hidden_size when necessary. I've also run the linter (black and others), and the issue should now be resolved. Let me know if you need any further adjustments or checks!

**Comment by Ying1123** [comment]
Hi @mssongit, thanks for the contribution! Could you provide a lora adapter path for testing?

File: python/sglang/srt/lora/lora_manager.py [hunk 1]
```diff
@@ -104,12 +104,18 @@ def init_loras(self):
...
                 for module in self.origin_target_modules
-            ]
-        )
+            }
+        else:
+            named_modules_dict = dict(self.base_model.named_modules())
+            self.target_modules = {
+                name
+                for name, module in named_modules_dict.items()
+                if module in self.origin_target_modules
```
Comment (Ying1123) [inline python/sglang/srt/lora/lora_manager.py]: This does not seem correct. `self.origin_target_modules` is a list of strings, not modules. You can refer to the `get_module_name` function in `llama.py`.

**Comment by mssongit** [comment]
> Hi @mssongit, thanks for the contribution! Could you provide a lora adapter path for testing?

The base model I am using is Qwen/Qwen2.5-14B-Instruct, and the LoRA adapter has been uploaded to mssongit/Qwen2.5-14B-SFT-LoRA. Feel free to use that for testing!

File: python/sglang/srt/lora/lora_manager.py (hunk 1 above)
Comment (mssongit) [reply python/sglang/srt/lora/lora_manager.py]: Thank you for the explanation. Based on your feedback, I have updated the commit accordingly and submitted it again.

Comment (Ying1123) [reply python/sglang/srt/lora/lora_manager.py]: Updated.

**Comment by Ying1123** [comment]
Hi @mssongit, I have directly modified your PR so that you may take a look. Currently, there is some issue with the base model "Qwen/Qwen2.5-14B-Instruct", which I'm still investigating. The fix for the base model will come up in another PR.

**Pull request merged**